Add support for Windows Mixed Reality controllers

CheesecakeCG opened the pull request:
I've only added support for the original WMR controllers so far, and I haven't much tested outside of VRChat and ChilloutVR.

~~Also, the grip poses in Monado might be flipped... and the ones I'm using here are rotated wrong.~~
The offsets are now fixed in Beyley's `rift-driver-tracking` branch and Thaytan's `dev-constellation-controller-tracking` branch of Monado.

- [x] OG WMR Controller Support (mostly done)
- [x] Samsung Odyssey Controller Support
- [X] HP Reverb G2 Controller Support

I'll need other people to test the last two controllers.

Supreeeme commented:
Feel free to split odyssey and reverb g2 support into separate PRs, if you'd like.

Supreeeme commented on src/input/profiles.rs:
```diff
@@ -1,14 +1,21 @@
     action_manifest::ControllerType, legacy::LegacyBindings, skeletal::SkeletalInputBindings,
 };
-use crate::openxr_data::Hand;
+use crate::{
+    input::profiles::samsung_odyssey_controller::SamsungOdysseyController, openxr_data::Hand,
+};
```
```suggestion
use crate::openxr_data::Hand;
use samsung_odyssey_controller::SamsungOdysseyController;
```

Supreeeme commented on src/input/profiles.rs:
```diff
@@ -109,6 +116,15 @@ impl Profiles {
         // Add supported interaction profiles here.
         static P: Profiles = Profiles {
             list: &[
+                (ControllerType::HPMotionController, &ReverbG2Controller),
+                (
+                    ControllerType::SamsungOdysseyController,
+                    &SamsungOdysseyController,
+                ),
```
Since these controller types are locked behind extensions, we will need to handle the extensions not being available in some way.

Supreeeme replied:
Now merged!

Supreeeme reviewed (commented):
These could use some simple `verify_bindings` tests. You can look at the ones in the other profiles for inspiration.

toyowheelin commented:
I am not sure if this is a good place for this or not but I built Cheesecakes wmr-controller-support branch and tested it with Xplane11 and ended up with a crash. I was in communication with Supreme in Discord and they were thinking it was likely that Cheesecakes branch just needs to be rebased. I would love to test and am hopeful that Cheesecake will see this and be able to do that. For reference this is the log from the crash https://bin.garudalinux.org/?d2bb0dfa11ef9f74#7uRhUQbjF928BVyHuLm8km9aXs7Cp8WRc6LBzE2YspYC

toyowheelin commented:
Unfortunately the rebase didn't help and resulted in the same error. https://bin.garudalinux.org/?11a46fbeba5363eb#BAPbEdpGxLu6gvtc8L7AZ5KZQaoYLyPU7XiYnNJQuG6 I will try to compile the default xrizer and make sure it works, as I have only ever used OpenComposite with Xplane11.

CheesecakeCG commented:
@toyowheelin Which kind of controllers are you using? Reverb G2?

CheesecakeCG commented:
> Let me know if you need any more info to look into this, I would love to help make this work.

I still need the render_model_name for the Samsung controllers, unless SteamVR doesn't actually show different models for those controllers. I'm not entirely sure where in the SteamVR driver that's defined though, I'm able to see it in the other drivers but not for WMR.

Commit:
Add support for Windows Mixed Reality controllers

Add basic support for first generation Windows Mixed Reality controllers

Enable Samsung Odyssey Controller and HP Controller extensions if they're available.

Add Samsung Odyssey controllers

Added tests (WIP)

Corrected values based on official drivers

Check for required extensions with WMR

Use escaped backslashes for Windows style path in WMR render_model_name

Formatting fixes

Get all tests to pass (non-functional)

Fixed botched rebase

Commit:
Fix HP Motion Controllers having a clicking grip button instead an
analog one

Commit:
Added comment about issue with Samsung Odyssey Controllers, and fixed tests to match.

Commit:
Make SamsungOdysseyController use ControllerType::HolographicController

Commit:
Removed redundant static lifetimes

Supreeeme commented:
Sorry, I haven't mentioned it, but I overhauled how input profiles are implemented. Hopefully not too much work to rebase.
